programs/gzip.c

```diff
@@ -258,7 +260,13 @@ do_decompress(struct libdeflate_decompressor *decompressor,
 			goto out;
 		}
 
-		ret = full_write(out, uncompressed_data, actual_out_nbytes);
+		if (out != NULL) {
+			ret = full_write(out, uncompressed_data, actual_out_nbytes);
+		}
+		else { /// NULL means test mode: don't write output
+			ret = uncompressed_size; /// as do_decompress() finished successfully it means file was decompressed fine
```

Comment:
This doesn't make sense; this is making the program always fail (exit with failure status) when `-t` was specified.

I recommend passing the `options` to this function.  Then it would be pretty self-explanatory to write:

                                                                                 
                if (!options->test) {                                            
                        ret = full_write(out, uncompressed_data,                 
                                         actual_out_nbytes);                     
                        if (ret != 0)                                            
                                goto out;                                        
                }                                                                


In addition to fixing this, can you please add test cases to `scripts/gzip_tests.sh` to test that this new option works correctly?

Reply:
I didn't know what exactly is to be returned.
That's also why I included diff patch rather than pull request.